### PR TITLE
Structured logging

### PR DESCRIPTION
**Pull request by dahlia**
Gives logs more context and make them to be paired so that beginning and ending of stages are easy to be analyzed.  I indeed analyzed logs using [a Python script][1].  Also replaced `Nito.AsyncEx.AsyncCollection<T>` with `System.Threading.Channels.Channel<T>` which is in the standard library.

[1]: https://gist.github.com/dahlia/8075b516b770a03c713ecd9b7cf2925b

### CODE REVIEW

File: Libplanet/Net/Transports/NetMQTransport.cs [hunk 1]
```diff
@@ -727,30 +729,35 @@ private void DoReply(object sender, NetMQQueueEventArgs<NetMQMessage> e)
...
+                        _requests.Writer.TryWrite(req.Retry());
                         await Task.Delay(retryAfter, cancellationToken);
                     }
                     else
                     {
                         _logger.Error("Failed to process request[{req}]; discard it.", req);
                     }
                 }
+
+#if NETCOREAPP3_0 || NETCOREAPP3_1 || NET
```
Comment (longfin) [inline Libplanet/Net/Transports/NetMQTransport.cs:741]: Is this guard intended?

**Review by longfin — DISMISSED** [review]

File: Libplanet/Net/Transports/NetMQTransport.cs (hunk 1 above)
Comment (dahlia) [reply Libplanet/Net/Transports/NetMQTransport.cs:741]: Yes, it's intended.

**Comment by dahlia** [comment]
@planetarium/libplanet PTAL.

**Review by riemannulus — APPROVED** [review]

**Comment by dahlia** [comment]
@planetarium/libplanet Rebased; please take a look again.

**Review by riemannulus — DISMISSED** [review]

**Review by longfin — DISMISSED** [review]

**Review by limebell — APPROVED** [review]

**Review by moreal — APPROVED** [review]

File: Libplanet/Net/Swarm.MessageHandlers.cs [hunk 2]
```diff
@@ -150,7 +159,8 @@ private void ProcessMessageHandler(object target, Message message)
                         header.Index,
                         ByteUtil.Hex(header.Hash),
                         peer);
-                    BlockDemand = new BlockDemand(header, peer, DateTimeOffset.UtcNow);
+                    var blockDemand = new BlockDemand(header, peer, DateTimeOffset.UtcNow);
+                    BlockDemand = blockDemand;
```
[the remark this answers is not in the document]
Comment (dahlia) [reply Libplanet/Net/Swarm.MessageHandlers.cs:163]: Same.  It seems left after I had logged this value to investigate things and cleaned up logs.

[the remark this answers is not in the document]
Comment (moreal) [reply Libplanet/Net/Swarm.MessageHandlers.cs:163]: Then, it seems able to be removed from the changes. How do you think?

[the remark this answers is not in the document]
Comment (dahlia) [reply Libplanet/Net/Swarm.MessageHandlers.cs]: Amended: 25c0027cc2fcb352bbebccc346dd44333a8e94b7.

**Comment by dahlia** [comment]
The build is now all green, and this PR had been approved before it's rebased due to a [small change](https://github.com/planetarium/libplanet/pull/1241#discussion_r612956549), so I'm going to merge this.

**Pull request merged**